torch/_inductor/optimize_indexing.py

```diff
@@ -18,8 +18,12 @@
 
 @dataclasses.dataclass(frozen=True)
 class ValueRanges(object):
-    lower: Union[sympy.Symbol, sympy.Number, int, float, bool]
-    upper: Union[sympy.Symbol, sympy.Number, int, float, bool]
+    lower: Union[sympy.Expr, sympy.Number, int, float, bool]
+    upper: Union[sympy.Expr, sympy.Number, int, float, bool]
+
+    def __contains__(self, x):
```

Comment:
This might be a bit of a confusing api, because it's not necessarily going to return True or False as is the typical convention, it might return a sympy expression. maybe assert that the output is not an expr?